### PR TITLE
SQLite engine added

### PR DESCRIPTION
**Pull request by Fusion**
Hello. Just added SQLite as an alternative to using DynamoDb.

It's for those folks who, like me, want to use this for personal projects without too much setup work :)

You will notice that a new directory, sql/, contains a couple scripts to add a subdomain reservation, create a pseudo-account and generate a hashed key,

Note that I write Rust code about once a year, so feel free to point out how terrible everything is!

Cheers.

### CODE REVIEW

File: tunnelto_server/src/auth_db.rs [hunk 1]
```diff
@@ -77,53 +122,92 @@ impl AuthDbService {
...
+                    params![auth_key_hash,],
+                    |row| row.get(0)
+                );
+                match row {
+                    Ok(value) => value,
+                    Err(_) => "default".to_string()
+                }
+            },
+            (Engine::DYNAMODB, None) => {
+                "default".to_string()
```
[the remark this answers is not in the document]
Comment (Fusion) [reply tunnelto_server/src/auth_db.rs:159]: I will look into cargo features. Good point about dependencies.

I will also look into your other comments :)

File: tunnelto_server/src/auth_db.rs [hunk 2]
```diff
@@ -7,18 +7,63 @@ use thiserror::Error;
...
+    engine: Engine,
+    client: Option<DynamoDbClient>,
 }
 
 impl AuthDbService {
     pub fn new() -> Result<Self, Box<dyn std::error::Error>> {
         let provider = EnvironmentProvider::default();
         let http_client = HttpClient::new()?;
         let client = Client::new_with(provider, http_client);
+        let engine = std::env::var("STORE").unwrap_or("dynamodb".to_string());
```
Comment (agrinman) [inline tunnelto_server/src/auth_db.rs:41]: My preference would be to gate the sqlite as a feature and then provide a common interface instead of the "if = " approach in each of the handlers.

Comment (Fusion) [reply tunnelto_server/src/auth_db.rs:41]: OK. Now a conditional feature.

File: tunnelto_server/src/auth_db.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Fusion) [reply tunnelto_server/src/auth_db.rs:159]: Got rid of this branch.

File: tunnelto_server/src/auth_db.rs [hunk 3]
```diff
@@ -77,53 +122,92 @@ impl AuthDbService {
...
+                let conn = Connection::open("./tunnelto.db").expect("Unable to open database for authentication purpose");
+                let row: Result<String, _> = conn.query_row(
+                    "SELECT account_id FROM tunnelto_auth WHERE auth_key_hash=?",
+                    params![auth_key_hash,],
+                    |row| row.get(0)
+                );
+                match row {
+                    Ok(value) => value,
+                    Err(_) => "default".to_string()
+                }
```
[the remark this answers is not in the document]
Comment (Fusion) [reply tunnelto_server/src/auth_db.rs:156]: Absolutely, 100% correct.

**Comment by Fusion** [comment]
So...I now have a fixed version. Did you close the request for any particular reason other than the fact that I took a very long time to get around to fixing it? ;)
If not, please re-open it so that I can push my new, revised atrocity.

**Comment by agrinman** [comment]
Hi there -- please take a look at: https://github.com/agrinman/tunnelto/blob/master/tunnelto_server/src/auth/mod.rs#L54
Any auth backend should implement this trait and then be configurable with a feature (see the lazy_static main.rs).
Instead of feature separation inside each function (which is hard to follow/read/debug) this should be a separate module altogether that is feature-gated.

**Comment by Fusion** [comment]
I see you guys haven't been sitting our your hands :)

OK I'll pull the latest code and work on that.
